### DISCUSSION TITLE
关于该课程的后续

### DISCUSSION THREAD

**Discussion by hbsun2113** [mlc-ai/mlc-zh#140] [Q&A]
课程结束了。首先，非常感谢tianqi老师以及@Hzfengsy @yzh119的多次帮助！其次，就第10次课程中tianqi的总结，我想提问3个问题，欢迎助教们和社区中的同学们提供建议：
1. 如何将本课程所学内容进行应用？tianqi在视频中粗略讲了两个应用场景，大家还有其他比较**具体**的任务或者场景，对所学内容进行练习吗？
2. 如何在本课程基础上深入学习MLC？我理解本课程最核心的目的是带大家了解MLC的总纲，因此没有涉及太广和更深入的细节实现。大家后续准备如何深入学习？
3. 课程结束后，这个论坛还会被继续维护吗？会承担其他任务吗？

**Comment by yzh119** [comment]
如果想深入了解实现细节的话，可以参考如下内容：
1. TensorIR的设计（[头文件](https://github.com/apache/tvm/blob/da48e13b66fa053578815343c3f247f47364d0bb/include/tvm/tir/)，[实现](https://github.com/apache/tvm/tree/da48e13b66fa053578815343c3f247f47364d0bb/src/tir/ir)）.
  1. [transformation的实现](https://github.com/apache/tvm/tree/da48e13b66fa053578815343c3f247f47364d0bb/src/tir/transforms)
  2. [schedule的实现](https://github.com/apache/tvm/tree/da48e13b66fa053578815343c3f247f47364d0bb/src/tir/schedule) （schedule可以当作带参数的transformation）
2. [Relax的设计文档](https://github.com/tlc-pack/relax/wiki)
3. [MLIR的各种方言](https://mlir.llvm.org/docs/Dialects/)

其中TensorIR和Relax的源码中广泛使用TVM的object system，可以参考[文档](https://tvm.apache.org/docs/arch/runtime.html)。
